### DISCUSSION TITLE
Directly map source `object` to target `array` since `v1.1.0`

### DISCUSSION THREAD

**Discussion by ToshY** [rekalogika/mapper#43] [Q&A] [answered]
I have an object that I want to convert back to an array. Currently I've found that passing `(object)[]` or `new \stdClass()` to the target, and then casting the result with `(array)` does the trick:

```php
$book = new Book(id: 1, title: 'Book Title');

$result = (array) $this->mapper->map($book, new \stdClass());

// ['id' => 1, 'title' => 'Book Title'];
```

My question: is this the proper way of using the mapper, or can it be done another/better way?

---

Read in the README that since `v1.1.0` the text [`Maps an object to an array, and vice versa.`](https://github.com/rekalogika/mapper?tab=readme-ov-file#general) was added, so if you have an example on how this can be achieved with the mapper would be much appreciated (as I also could not find an example in the [docs](https://rekalogika.dev/mapper/object-array)).

---

**Version**: `v1.1.0`

**Comment by priyadi** [comment, marked as answer]
As of 1.1, the mapper is fully capable of handling arrays as mentioned in the documentation. Thing is that array is supported, only not as the root object.

The problem is now we are bound to `MapperInterface`. I decided on `map()` taking an object and returning an object to simplify the job of those who want to extend the mapper.

In the future, I'll probably create something like `MixedMapperInterface` that accepts and returns mixed. But I still have to figure out how to specify array as the target, do we use the string `'array'` or specify an empty array `[]`? Then there is a question on the use cases where the source is a collection, and we want to map it to `array<int,Something>`

I think I'll defer these decisions until after I migrate mapper to use `symfony/type-info` where they have a better syntax to specify types.

In the meantime, I think you should create your own adapter for your use case. It would convert an array input to `stdClass`, then pass it to mapper, then if the output is an `stdClass`, your adapter would convert it back to array.

**Reply by ToshY** [reply]
I understand the decision and the suggestion to make a custom adapter for this use case is very reasonable. Thank you for taking your time to answer, very much appreciated. 🙂
